src/components/MultiHopTrade/components/VerifyAddresses/VerifyAddresses.tsx

```diff
@@ -163,13 +158,64 @@ export const VerifyAddresses = () => {
     fetchAddresses()
   }, [fetchAddresses])
 
-  const handleVerify = useCallback(
-    async (type: 'sell' | 'buy') => {
+  const isVerifying = useMemo(
+    () => isSellVerifying || isBuyVerifying,
+    [isSellVerifying, isBuyVerifying],
+  )
+
+  const setIsVerifying = useCallback(
+    (type: 'sell' | 'buy', isVerifying: boolean) => {
+      if (type === 'sell') {
+        setIsSellVerifying(isVerifying)
+      } else if (type === 'buy') {
+        setIsBuyVerifying(isVerifying)
+      }
+    },
+    [setIsSellVerifying, setIsBuyVerifying],
+  )
+
+  const setErrorMessage = useCallback((type: 'sell' | 'buy', errorMessage: string) => {
+    if (type === 'sell') {
+      setSellErrorMessage(errorMessage)
+    } else if (type === 'buy') {
+      setBuyErrorMessage(errorMessage)
+    }
+  }, [])
+
+  const clearErrorMessage = useCallback((type: 'sell' | 'buy') => {
+    if (type === 'sell') {
+      setSellErrorMessage(undefined)
+    } else if (type === 'buy') {
+      setBuyErrorMessage(undefined)
+    }
+  }, [])
+
+  const setStatus = useCallback(
+    (type: 'sell' | 'buy', status: 'pending' | 'verified' | 'error' | 'notRequired') => {
```

Comment:
Can we make `status` an enum type and then use it throughout this file? E.g. `TradeStatus.Pending` etc

Ideally `type` would be too, though not a specific concern of this PR.

Reply:
https://github.com/shapeshift/web/pull/7639/commits/ffcadbf199a3b8f15fcbe5e8199b5fb1985c2fe4